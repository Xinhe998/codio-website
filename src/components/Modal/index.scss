@import '../../../styles/main.scss';

$pc-media: 960px;
$pad-media: 760px;
$phone-media: 480px;
.Modal__backdrop {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  // width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 100;
}
.Modal__wrapper {
  background-color: #ffffff;
  border-radius: 10px;
  max-width: 800px;
  // min-height: 300px;
  width: 550px;
  overflow: hidden;
  transition: all 0.5s;
  transform: scale(0.5);
  -webkit-animation: ScaleUp 0.3s; /* Safari, Chrome and Opera > 12.1 */
  -moz-animation: ScaleUp 0.3s; /* Firefox < 16 */
  -ms-animation: ScaleUp 0.3s; /* Internet Explorer */
  -o-animation: ScaleUp 0.3s; /* Opera < 12.1 */
  animation: ScaleUp 0.3s;
  animation-fill-mode: forwards;
  @media screen and (max-width: $pad-media) {
    width: 100%;
  }
}
.closing .Modal__wrapper {
  -webkit-animation: ScaleDown 0.3s; /* Safari, Chrome and Opera > 12.1 */
  -moz-animation: ScaleDown 0.3s; /* Firefox < 16 */
  -ms-animation: ScaleDown 0.3s; /* Internet Explorer */
  -o-animation: ScaleDown 0.3s; /* Opera < 12.1 */
  animation: ScaleDown 0.3s;
  animation-fill-mode: forwards;
}
.Modal__content {
  height: calc(100% - 108px);
  transition: all 0.5s;
  padding: 30px 50px 50px;
}
.Modal__title {
  height: 54px;
  padding: 15px 20px 15px 50px;
  display: flex;
  align-items: center;
  user-select: none;
  font-size: $modal-title-font-size;
  line-height: 1.1;
  color: $modal-title-color;
  background-color: $modal-header-background-color;
  svg {
    width: 30px;
    height: 30px;
    margin-left: auto;
    cursor: pointer;
    color: #3597ec;
  }
}
.Modal__button-group {
  height: 54px;
  display: flex;
  user-select: none;
  box-shadow: 0 -4px 10px 0 rgba(0, 0, 0, 0.1);
}

@keyframes ScaleUp {
  from {
    transition: transform 300ms ease-in;
    transform: scale(0.5);
  }
  to {
    transform: scale(1);
  }
}

@keyframes ScaleDown {
  from {
    transition: transform 300ms ease-in;
    transform: scale(1);
  }
  to {
    transform: scale(0.5);
  }
}
