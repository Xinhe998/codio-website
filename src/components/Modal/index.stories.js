import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { DocsPage, DocsContainer } from '@storybook/addon-docs/blocks';
import StoryRouter from 'storybook-react-router';
import Modal from './index';
import storeFs from '../../reducers';

const store = createStore(storeFs);
const stories = storiesOf('Modal', module).addParameters({
  title: 'Modal',
  component: Modal,
  docs: {
    container: DocsContainer,
    page: DocsPage,
  },
});

stories.addDecorator(StoryRouter()).add('default', () => (
  <Provider store={store}>
    <Modal
      isOpen
      title="My PopUp Modal"
      onClose={() => {}}
      shouldCloseOnEsc
      shouldCloseOnClickOutside
      showControlBtn={false}
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industry's standard dummy text ever since the
      1500s, when an unknown printer took a galley of type and scrambled it to
      make a type specimen book. It has survived not only five centuries, but
      also the leap into electronic typesetting, remaining essentially
      unchanged. It was popularised in the 1960s with the release of Letraset
      sheets containing Lorem Ipsum passages, and more recently with desktop
      publishing software like Aldus PageMaker including versions of Lorem
      Ipsum.
    </Modal>
  </Provider>
));

stories.addDecorator(StoryRouter()).add('顯示control buttons', () => (
  <Provider store={store}>
    <Modal
      isOpen
      title="My PopUp Modal"
      onClose={() => {}}
      shouldCloseOnEsc
      shouldCloseOnClickOutside
      showControlBtn
      confirmBtnText="確定"
      cancelBtnText="取消"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industry's standard dummy text ever since the
      1500s, when an unknown printer took a galley of type and scrambled it to
      make a type specimen book. It has survived not only five centuries, but
      also the leap into electronic typesetting, remaining essentially
      unchanged. It was popularised in the 1960s with the release of Letraset
      sheets containing Lorem Ipsum passages, and more recently with desktop
      publishing software like Aldus PageMaker including versions of Lorem
      Ipsum.
    </Modal>
  </Provider>
));
