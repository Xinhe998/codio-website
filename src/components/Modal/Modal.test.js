/* eslint-disable func-names */
import 'babel-polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, configure, render, mount } from 'enzyme';
import { createMemoryHistory } from 'history';
import Adapter from 'enzyme-adapter-react-16';
import Modal from './index';

import storeFs from '../../reducers';

const store = createStore(storeFs);

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
configure({ adapter: new Adapter() });

describe('Modal', function() {
  let testComp;

  beforeEach(() => {
    jest.resetModules();
    this.params = {
      isOpen: true,
      title: 'TestModal',
      shouldCloseOnEsc: true,
      shouldCloseOnClickOutside: true,
      showControlBtn: true,
      cancelBtnText: '取消',
      confirmBtnText: '確認',
      disabled: false,
      onClose: jest.fn(),
      Confirm: jest.fn(),
      children: <h1>Test</h1>,
      history: createMemoryHistory('/'),
      className: 'TestModal',
    };

    this.makeSubject = () => {
      const {
        isOpen,
        title,
        shouldCloseOnEsc,
        shouldCloseOnClickOutside,
        showControlBtn,
        cancelBtnText,
        confirmBtnText,
        disabled,
        onClose,
        Confirm,
        children,
        history,
        className,
      } = this.params;

      testComp = shallow(
        <Router>
          <Provider store={store}>
            <Modal
              isOpen={isOpen}
              title={title}
              shouldCloseOnEsc={shouldCloseOnEsc}
              shouldCloseOnClickOutside={shouldCloseOnClickOutside}
              showControlBtn={showControlBtn}
              cancelBtnText={cancelBtnText}
              confirmBtnText={confirmBtnText}
              disabled={disabled}
              onClose={onClose}
              Confirm={Confirm}
              history={history}
              className={className}
            >
              {children}
            </Modal>
          </Provider>
        </Router>,
      );
      return testComp;
    };
  });
  describe('when it has complete data', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });
    it('should be same as snapshot', async () => {
      const wrapper = render(await testComp);
      expect(wrapper).toMatchSnapshot();
    });
    it('should find className', async () => {
      expect(testComp.find('.Modal'));
      expect(testComp.find('.TestModal'));
    });
  });
});
