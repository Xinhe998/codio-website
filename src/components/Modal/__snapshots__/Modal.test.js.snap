// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal when it has complete data should be same as snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "TestModal Modal Modal__backdrop",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "Modal__wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "Modal__title",
            },
            "children": Array [
              Object {
                "data": "TestModal",
                "next": Object {
                  "attribs": Object {
                    "class": "close_btn",
                    "fill": "currentColor",
                    "height": "1em",
                    "stroke": "currentColor",
                    "stroke-width": "0",
                    "viewBox": "0 0 512 512",
                    "width": "1em",
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "http://www.w3.org/2000/xmlns/",
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "",
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
              Object {
                "attribs": Object {
                  "class": "close_btn",
                  "fill": "currentColor",
                  "height": "1em",
                  "stroke": "currentColor",
                  "stroke-width": "0",
                  "viewBox": "0 0 512 512",
                  "width": "1em",
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "data": "TestModal",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                  "xmlns": "http://www.w3.org/2000/xmlns/",
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "fill": undefined,
                  "height": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                  "xmlns": "",
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "Modal__content",
              },
              "children": Array [
                Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Test",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "Modal__button-group",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "Modal__button-group__cancel",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "取消",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "class": "Modal__button-group__confirm",
                            "type": "button",
                          },
                          "children": Array [
                            Object {
                              "data": "確認",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "button",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                            "type": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                            "type": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {
                          "class": "Modal__button-group__confirm",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "確認",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "class": "Modal__button-group__cancel",
                            "type": "button",
                          },
                          "children": Array [
                            Object {
                              "data": "取消",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "button",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                            "type": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                            "type": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                Object {
                  "attribs": Object {
                    "class": "Modal__button-group",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "Modal__button-group__cancel",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "取消",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "class": "Modal__button-group__confirm",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "確認",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "class": "Modal__button-group__confirm",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "確認",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "Modal__button-group__cancel",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "取消",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "Test",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h1",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "Modal__content",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Test",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "Modal__button-group",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "Modal__button-group__cancel",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "取消",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "class": "Modal__button-group__confirm",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "確認",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "class": "Modal__button-group__confirm",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "確認",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "Modal__button-group__cancel",
                          "type": "button",
                        },
                        "children": Array [
                          Object {
                            "data": "取消",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "button",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "type": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              Object {
                "attribs": Object {
                  "class": "Modal__button-group",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "Modal__button-group__cancel",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "data": "取消",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "class": "Modal__button-group__confirm",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "確認",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "class": "Modal__button-group__confirm",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "data": "確認",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "class": "Modal__button-group__cancel",
                        "type": "button",
                      },
                      "children": Array [
                        Object {
                          "data": "取消",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Test",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "Modal__title",
              },
              "children": Array [
                Object {
                  "data": "TestModal",
                  "next": Object {
                    "attribs": Object {
                      "class": "close_btn",
                      "fill": "currentColor",
                      "height": "1em",
                      "stroke": "currentColor",
                      "stroke-width": "0",
                      "viewBox": "0 0 512 512",
                      "width": "1em",
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                      "xmlns": "http://www.w3.org/2000/xmlns/",
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "fill": undefined,
                      "height": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                      "viewBox": undefined,
                      "width": undefined,
                      "xmlns": "",
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
                Object {
                  "attribs": Object {
                    "class": "close_btn",
                    "fill": "currentColor",
                    "height": "1em",
                    "stroke": "currentColor",
                    "stroke-width": "0",
                    "viewBox": "0 0 512 512",
                    "width": "1em",
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "d": "M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "data": "TestModal",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "http://www.w3.org/2000/xmlns/",
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "fill": undefined,
                    "height": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                    "xmlns": "",
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
